internal/controller/serverset/serverset.go

```diff
@@ -174,16 +174,52 @@ func (e *external) populateReplicasStatuses(ctx context.Context, cr *v1alpha1.Se
 			errMsg = lastCondition.Message
 		}
 
+		replicaIdx := ComputeReplicaIdx(fmt.Sprintf(indexLabel, cr.Name, ResourceServer), serverSetReplicas[i].Labels)
+		nicStatues, _ := computeNicStatuses(ctx, e, cr.Name, replicaIdx)
 		cr.Status.AtProvider.ReplicaStatuses[i] = v1alpha1.ServerSetReplicaStatus{
 			Role:         fetchRole(ctx, e, serverSetReplicas[i]),
 			Name:         serverSetReplicas[i].Name,
+			Index:        replicaIdx,
+			NICStatuses:  nicStatues,
 			Status:       replicaStatus,
 			ErrorMessage: errMsg,
 			LastModified: metav1.Now(),
 		}
 	}
 }
 
+func computeNicStatuses(ctx context.Context, e *external, crName string, replicaIndex int) ([]v1alpha1.ServerSetNICStatus, error) {
+	nics, err := GetNICsOfSSet(ctx, e.kube, crName)
```

Comment:
can't we use ListResFromSSetWithIndex instead, to not need keepOnlyNicsOfReplica?

Reply:
done